app/code/Magento/InventorySourceSelectionApi/Model/InventoryRequestFromOrderBuilderInterface.php

```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventorySourceSelectionApi\Model;
+
+use Magento\InventorySourceSelectionApi\Api\Data\InventoryRequestInterface;
+use Magento\InventorySourceSelectionApi\Api\Data\ItemRequestInterface;
+
+/**
+ * Interface InventoryRequestFromOrderBuilderInterface
+ *
+ * @api
+ */
+interface InventoryRequestFromOrderBuilderInterface
+{
+    /**
+     * Create an inventory request from one order and a set of items
+     *
+     * @param int $stockId
+     * @param int $orderId
+     * @param ItemRequestInterface[] $requestItems
+     * @return InventoryRequestInterface
+     */
+    public function execute(int $stockId, int $orderId, array $requestItems): InventoryRequestInterface;
```

Comment:
This interface looks like a wrong abstraction.
Even currently, having just two implementations: Priority Based algorithm and Distance-based
the default implementation does not use `$orderId` at all
```
     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
     */
    public function execute(int $stockId, int $orderId, array $requestItems): InventoryRequestInterface
    {
        $inventoryRequest = $this->inventoryRequestFactory->create([
            'stockId' => $stockId,
            'items'   => $requestItems
        ]);

        return $inventoryRequest;
    }
```

What will we do in the future if we would need to have New Algorithm which works with `Quote` object? 
So that we have to pass a Quote Id as a parameter.


My point here that we no need to introduce an Interface for this Builder which combines input data for all cases. Because these cases are different and each should have own interface

Reply:
This class is intended to be used to build a request from an order, this is why it is called `InventoryRequestFromOrderBuilderInterface`, the main idea behind is that we could create different builders depending on the source.

If we need to create a request from a quote we should use something like `InventoryRequestFromQuoteBuilderInterface`. This was one of the point we discussed at MLEU.

Does it make sense?